### PR TITLE
feat: dataset and versioneddataset impl

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: dataset and versioneddataset impl

#### Which issue(s) this PR fixes

Fixes #

### CODE REVIEW

**Review by bjwswang — CHANGES_REQUESTED** [review]
fix the golangci-lint errors

File: api/v1alpha1/dataset.go [hunk 1]
```diff
@@ -29,5 +29,6 @@ var (
 	// LabelDatasetScene defines the content type of this dataset
 	LabelDatasetContentType = Group + "/content-type"
 	// LabelDatasetBestCase defines the best case to use this dataset
-	LabelDatasetBestCase = Group + "/best-case"
+	LabelDatasetBestCase  = Group + "/best-case"
+	LabelDatasetFinalizer = Group + "/finalizers"
```
Comment (bjwswang) [inline api/v1alpha1/dataset.go:33]: we have a common finalizer label at https://github.com/kubeagi/arcadia/blob/main/api/v1alpha1/common.go#L25

File: api/v1alpha1/versioneddataset.go [hunk 2]
```diff
@@ -16,6 +16,144 @@ limitations under the License.
...
+			fileGroup[key] = make([]string, 0)
+		}
+		fileGroup[key] = append(fileGroup[key], fg.Paths...)
+	}
+
+	// 3. Convert fileGroup to []DatasourceFileStatus format
+	targetDatasourceFileStatus := make([]DatasourceFileStatus, 0)
+	var namespace, name string
+	for datasource, filePaths := range fileGroup {
+		_, _ = fmt.Sscanf(datasource, "%s %s", &namespace, &name)
```
Comment (bjwswang) [inline api/v1alpha1/versioneddataset.go:65]: @0xff-dev what is this line for ?

Comment (0xff-dev) [reply api/v1alpha1/versioneddataset.go:65]: Get namespace and name.

**Review by bjwswang — APPROVED** [review]

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**